ocs_ci/deployment/flexy.py

```diff
@@ -0,0 +1,274 @@
+
+    def build_destroy_cmd(self):
+        """
+        Build flexy command line for 'destroy' operation
+
+        """
+        cmd = "sudo podman run --rm=true"
+        flexy_container_args = self.build_container_args()
+        cmd_string = " ".join([cmd, " ".join(flexy_container_args), 'destroy'])
+        return cmd_string
+
+    def get_log(self):
+        """
+        This function runs in a seperate thread and continuously reads
+        logs from flexy podman
+
+        """
+        for res in TimeoutSampler(20, 1, os.path.exists, self.flexy_log_file):
+            if res:
+                break
+        log_fd = open(self.flexy_log_file, "r")
+
+        while True:
+            line = log_fd.readline()
+            if line:
+                logger.info(line.strip())
+            if self.exitThread:
+                break
+        logger.info("Logger thread done printing flexy logs")
+        # exit only this thread
+        sys.exit()
+
+    def build_container_args(self):
+        """
+        Builds most commonly used arguments for flexy container
+
+        """
+        args = list()
+        args.append(f"--log-opt={self.flexy_log_file}")
+        args.append(f"--env-file={self.flex_env_file}")
+        args.append(f"-w={self.flexy_mnt_container_dir}")
+        args.append(
+            f"--mount=type=bind,source={self.flexy_mnt_host_dir},"
+            f"destination=f{self.flexy_mnt_container_dir},relabel=shared"
+        )
+        args.append(
+            f"--mount=type=bind,source={self.flexy_host_private_conf_dir_path},"
+            f"destination={self.flexy_mnt_private_conf_dir},relabel=shared"
+        )
+        args.append(f"{self.flexy_img_url}")
+        return args
+
+    def clone_and_unlock_ocs_private_conf(self):
+        """
+        Clone ocs_private_conf (flexy env and config) repo into
+        flexy_mnt_host_dir
+
+        """
+        clone_repo(self.flexy_private_conf_url, self.flexy_host_private_conf_dir_path)
+        # git-crypt unlock /path/to/keyfile
+        old_cwd = os.getcwd()
+        os.chdir(self.flexy_host_private_conf_dir_path)
+        cp = subprocess.run(
+            f'git-crypt unlock {constants.FLEXY_GIT_CRYPT_KEYFILE}',
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE
+        )
+        if cp.returncode:
+            raise exceptions.CommandFailed(cp.stderr)
+        logger.info("Unlocked the git repo")
+        os.chdir(old_cwd)
+
+    def merge_flexy_env(self):
+        """
+        Update the ocs-osp.env file with the user supplied values.
+        This function assumes that flexy-ocs-private repo has been
+        already cloned
+
+        """
+        config_parser = configparser.ConfigParser()
+        # without this,config parser would convert all
+        # uppercase keys to lowercase
+        config_parser.optionxform = str
+        env_file = os.path.join(
+            self.flexy_host_private_conf_dir_path, constants.FLEXY_DEFAULT_ENV_FILE
+        )
+
```

Comment:
Couldn't this be omitted in favour of `self.flex_env_file` configured in `__init__()`?

Reply:
Sorry, I missed to notice that. I will change it.

Reply:
done